### PR TITLE
feat!: refactor contract calls and multicall to JS + fuel-asm

### PR DESCRIPTION
**Pull request by camsjams**
This PR converts the Sway file to asm, allowing for:
- a contract call script that doesn't require a Sway compile step
- easier upgrades
- smaller payload size for contract calls
- unlimited mulitcalls (currently 5 maximum)
- IMO the most important changes to look at are all within `packages/program/src/contract-call-script.ts`

### Notes:
- Uses fuel-asm-wasm from @xgreenx and @arboleya (thanks :tada: )
- Added basic wrapper for fuel-asm for DX ergonomics
- removes all usage and sources of multicall Sway contract
- added tests for multicalls greater than 5 now :rocket: 
- **10 file changes are just deletions**, many are just adjusting unit tests :+1: 
- reducing the size of the contract call script changed the gas required

### CODE REVIEW

File: packages/providers/src/transaction-request/script-transaction-request.ts [hunk 1]
```diff
@@ -80,8 +80,8 @@ export class ScriptTransactionRequest extends BaseTransactionRequest {
   }
 
   setScript<T>(script: AbstractScriptRequest<T>, data: T) {
-    this.script = script.bytes;
     this.scriptData = script.encodeScriptData(data);
+    this.script = script.bytes;
```
Comment (Dhaiwat10) [inline packages/providers/src/transaction-request/script-transaction-request.ts:84]: Any reason for this change in order? 😅

File: packages/program/src/contract-call-script.ts [hunk 2]
```diff
@@ -1,131 +1,204 @@
...
+  amountOffset: number;
+  assetIdOffset: number;
+};
+
+const DEFAULT_OPCODE_PARAMS = {
+  assetIdOffset: 0,
+  amountOffset: 0,
+  gasForwardedOffset: 0,
+  callDataOffset: 0,
+};
```
Comment (Dhaiwat10) [inline packages/program/src/contract-call-script.ts:38]:
```suggestion
const DEFAULT_OPCODE_PARAMS: CallOpcodeParamsOffset = {
  assetIdOffset: 0,
  amountOffset: 0,
  gasForwardedOffset: 0,
  callDataOffset: 0,
};
```
A small type annotation

**Review by Dhaiwat10 — DISMISSED** [review]
Super clean 💯

**Review by Dhaiwat10 — DISMISSED** [review]

**Review by Torres-ssf — DISMISSED** [review]

**Comment by Dhaiwat10** [comment]
@camsjams Looks like there are some merge conflicts that need to be resolved before this can be merged.

**Review by arboleya — CHANGES_REQUESTED** [review]
Can we use the [last version](https://www.npmjs.com/package/@fuels/vm-asm) of `@fuels/vm-asm`?

Version `0.36.0` includes that last fix to enforce [async initialization](https://github.com/FuelLabs/fuel-vm/pull/534):

```ts
import * as fuel_asm from '@fuels/vm-asm'

(async function() {
  await fuel_asm.initWasm();

  // fuel_asm.<?>();
  // ...
})();
```

**Review by arboleya — APPROVED** [review]
⚡

**Review by Dhaiwat10 — APPROVED** [review]

**Pull request merged**